packages/form-js-viewer/src/render/components/util/localisationUtil.js

```diff
@@ -0,0 +1,404 @@
+/**
+ * Returns date format for the provided locale.
+ * If the locale is not provided, uses the browser's locale.
+ *
+ * @param {string} [locale] - The locale to get date format for.
+ * @returns {string} The date format for the locale.
+ */
+export function getLocaleDateFormat(locale) {
+  const defaultFormat = 'dd/MM/yyyy';
+  const langCode = (locale || navigator.language).toLowerCase();
+  return countryFormats[langCode] || languageFormats[langCode.split('-')[0]] || defaultFormat;
+}
+
+/**
+ * Returns readable date format for the provided locale.
+ * If the locale is not provided, uses the browser's locale.
+ *
+ * @param {string} [locale] - The locale to get readable date format for.
+ * @returns {string} The readable date format for the locale.
+ */
+export function getLocaleReadableDateFormat(locale) {
+  let format = getLocaleDateFormat(locale).toLowerCase();
+
+  // Ensure year is in 'yyyy' format
+  if (format.includes('yy') && !format.includes('yyyy')) {
+    format = format.replace('yy', 'yyyy');
+  }
+
+  // Ensure month is in 'mm' format
+  if (format.includes('m') && !format.includes('mm')) {
+    format = format.replace('m', 'mm');
+  }
+
+  // Ensure day is in 'dd' format
+  if (format.includes('d') && !format.includes('dd')) {
+    format = format.replace('d', 'dd');
+  }
+
+  return format;
+}
+
+/**
+ * Returns flatpickr config for the provided locale.
+ * If the locale is not provided, uses the browser's locale.
+ *
+ * @param {string} [locale] - The locale to get flatpickr config for.
+ * @returns {object} The flatpickr config for the locale.
+ */
+export function getLocaleDateFlatpickrConfig(locale) {
+  return flatpickerizeDateFormat(getLocaleDateFormat(locale));
+}
+
+function flatpickerizeDateFormat(dateFormat) {
+
+  const useLeadingZero = {
+    day: dateFormat.includes('dd'),
+    month: dateFormat.includes('MM'),
+    year: dateFormat.includes('yyyy')
+  };
+
+  dateFormat = useLeadingZero.day ? dateFormat.replace('dd', 'd') : dateFormat.replace('d', 'j');
+  dateFormat = useLeadingZero.month ? dateFormat.replace('MM', 'm') : dateFormat.replace('M', 'n');
+  dateFormat = useLeadingZero.year ? dateFormat.replace('yyyy', 'Y') : dateFormat.replace('yy', 'y');
+
+  return dateFormat;
+}
+
+const countryFormats = {
```

Comment:
Why do we have these mappings if we're not using them?

Reply:
We do use them at line 11

Reply:
Basically, not everything is supported all the time, so we have specific mappings that I took from a source of international mappings (as a big dictionary instead of having yet another dependency), I also created a fallback dictionary for when only the country is specified, and then finally there's a global fallback to dd/mm/yyyy.

Reply:
Ah, sorry. I missed the `navigator.language` and only focused on the `locale`.
But anyway, I think we shouldn't keep this dictionary on `form-js`, very likely it's incomplete and it could easily get out of date form some locales.
I think we could use the browser `Intl` API to get the correct format (since we're already using it to get the locale/language)
We could try the functions below (slightly modified from StackOverflow):
```js
function getDateFormatString(lang = 'default') {
  const parts = new Intl.DateTimeFormat(lang).formatToParts(new Date());

  return parts
    .map(part => {
      switch (part.type) {
        case "day":
          return "DD";
        case "month":
          return "MM";
        case "year":
          return "YYYY";
        default:
          return part.value;
      }
    })
    .join("");
}
```